File: app/src/main/java/com/example/groxweatherapp/WeatherListAdapter.java
package com.example.groxweatherapp;

import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.RecyclerView.ViewHolder;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import com.example.groxweatherapp.model.Weather;
import java.util.ArrayList;
import java.util.List;

public class WeatherListAdapter extends RecyclerView.Adapter<WeatherListAdapter.WeatherListViewHolder> {

    private List<Weather> weatherList = new ArrayList<>();

    public void setWeatherList(List<Weather> weatherList) {
        this.weatherList.clear();
        this.weatherList.addAll(weatherList);
    }

    @Override
    public WeatherListViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.weather_item, parent, false);
        return new WeatherListViewHolder(view);
    }

    @Override
    public void onBindViewHolder(WeatherListViewHolder holder, int position) {
        Weather weather = weatherList.get(position);
        holder.date.setText(weather.date);
        holder.description.setText(weather.description);
        holder.temMax.setText(weather.temperatureMax);
        holder.tempMin.setText(weather.temperatureMin);
    }

    @Override
    public int getItemCount() {
        return weatherList.size();
    }

    public static class WeatherListViewHolder extends ViewHolder {

        @BindView(R.id.date) public TextView date;
        @BindView(R.id.description) public TextView description;
        @BindView(R.id.temp_max) public TextView temMax;
        @BindView(R.id.temp_min) public TextView tempMin;

        public WeatherListViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/com/example/groxweatherapp/WeatherListAdapter.kt
package com.example.groxweatherapp

import android.support.v7.widget.RecyclerView
import android.support.v7.widget.RecyclerView.ViewHolder
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import butterknife.ButterKnife
import com.example.groxweatherapp.WeatherListAdapter.WeatherListViewHolder
import com.example.groxweatherapp.model.Weather
import kotterknife.bindView
import java.util.*

class WeatherListAdapter : RecyclerView.Adapter<WeatherListViewHolder>() {

    private val weatherList = ArrayList<Weather>()

    fun setWeatherList(weatherList: List<Weather>) {
        this.weatherList.clear()
        this.weatherList.addAll(weatherList)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WeatherListViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.weather_item, parent, false)
        return WeatherListViewHolder(view)
    }

    override fun onBindViewHolder(holder: WeatherListViewHolder, position: Int) {
        val (date, temperatureMin, temperatureMax, description) = weatherList[position]
        holder.date.text = date
        holder.description.text = description
        holder.temMax.text = temperatureMax
        holder.tempMin.text = temperatureMin
    }

    override fun getItemCount(): Int {
        return weatherList.size
    }

    class WeatherListViewHolder(itemView: View) : ViewHolder(itemView) {

        val date: TextView by bindView(R.id.date)
        val description: TextView by bindView(R.id.description)
        val temMax: TextView by bindView(R.id.temp_max)
        val tempMin: TextView by bindView(R.id.temp_min)

        init {
            ButterKnife.bind(this, itemView)
        }
    }
}
